pkg/controllers/et/verticadbref_reconciler.go

```diff
@@ -0,0 +1,121 @@
+/*
+Copyright [2021-2023] Open Text.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package et
+
+import (
+	"context"
+	"fmt"
+
+	vapi "github.com/vertica/vertica-kubernetes/api/v1beta1"
+	"github.com/vertica/vertica-kubernetes/pkg/controllers"
+	"github.com/vertica/vertica-kubernetes/pkg/etstatus"
+	batchv1 "k8s.io/api/batch/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+)
+
+type VerticaDBRefReconciler struct {
+	VRec *EventTriggerReconciler
+	Et   *vapi.EventTrigger
+}
+
+func MakeVerticaDBRefReconciler(r *EventTriggerReconciler, et *vapi.EventTrigger) controllers.ReconcileActor {
+	return &VerticaDBRefReconciler{VRec: r, Et: et}
+}
+
+func (r *VerticaDBRefReconciler) Reconcile(ctx context.Context, req *ctrl.Request) (ctrl.Result, error) {
+	for _, ref := range r.Et.Spec.References {
+		if ref.Object.Kind != vapi.VerticaDBKind || ref.Object.APIVersion != vapi.GroupVersion.String() {
+			continue
+		}
+
+		vdb := &vapi.VerticaDB{}
+
+		nm := types.NamespacedName{
+			Namespace: r.Et.Namespace,
+			Name:      ref.Object.Name,
+		}
+
+		if err := r.VRec.Client.Get(ctx, nm, vdb); err != nil {
+			if errors.IsNotFound(err) {
+				continue
```

Comment:
We still need to update the status before we get here. Only part of the status needs to be updated. We won't have resourceVersion, UID, etc. But I want to see a status update as soon the reconciler runs. Even if there is no reference object created yet.

Reply:
I'm not sure I understand here

Reply:
I would like the status to be updated anytime the controller runs. It will be an indication that the controller ran and saw the ET. So, we need status updates for cases where we haven't yet trigger the job.